### PR TITLE
feat: add flag to CLI to disable colour coding of console output

### LINKED ISSUE #4030: Disable console color coding
### Describe the feature

Add cli flag that is going to disable the colour coding of console output, it is particularly annoying with docker logs that would contain ansci encoding.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by alessandromazza98**
Closes #4030 

This PR adds a flag `--log.not-coloured` to the CLI to disable the colour coding of console output.

### CODE REVIEW

File: bin/reth/src/cli/mod.rs
Comment (rakita) [inline bin/reth/src/cli/mod.rs:165]: @onbjerg you are better at naming things. Is this okay?

**Review by rakita — APPROVED** [review]
lgtm, pending @onbjerg

File: bin/reth/src/cli/mod.rs
Comment (mattsse) [inline bin/reth/src/cli/mod.rs:163]: we can adapt the docs from with_ansi here:

> Sets whether or not the formatter emits ANSI terminal escape codes for colors and other text formatting.

perhaps we should also rename the flag to `no-colors`

defer to @onbjerg here

**Review by mattsse — CHANGES_REQUESTED** [review]
docs nit,

otherwise pending @onbjerg

File: bin/reth/src/cli/mod.rs
Comment (alessandromazza98) [reply bin/reth/src/cli/mod.rs:163]: Ok I wait @onbjerg, thanks.

Comment (onbjerg) [reply bin/reth/src/cli/mod.rs:163]: It's way more common to see a flag named `--color` that accepts `auto`, `never` or `always`.

The auto mode usually does some environment detection, but for now it's ok if it just behaves as `always` imo

I'd prefer that to this flag

**Review by onbjerg — CHANGES_REQUESTED** [review]

File: bin/reth/src/cli/mod.rs
Comment (mattsse) [inline bin/reth/src/cli/mod.rs:171]: nit: lets make this an enum with a Fromstr impl

**Review by mattsse — CHANGES_REQUESTED** [review]

File: bin/reth/src/cli/mod.rs
Comment (alessandromazza98) [reply bin/reth/src/cli/mod.rs]: ok

**Review by mattsse — APPROVED** [review]
ty!

**Review by onbjerg — APPROVED** [review]

**Pull request merged**
